hibernate-core/src/main/java/org/hibernate/cfg/AnnotationBinder.java

```diff
@@ -3113,7 +3113,12 @@ else if (hasSpecjManyToOne) {
 		propertyBinder.setCascade( cascadeStrategy );
 		propertyBinder.setProperty( property );
 		propertyBinder.setXToMany( true );
-		propertyBinder.makePropertyAndBind();
+		if ( property.isAnnotationPresent( JoinTable.class ) ) {
+			propertyBinder.makePropertyAndBind().setOptional( optional );
+		}
+		else {
+			propertyBinder.makePropertyAndBind();
+		}
```

Comment:
I understand you wanna avoid a temporary variable for `makePropertyAndBind()` result, but it is still a little bit confusing why `propertyBinder.makePropertyAndBind()` remains the same across the two if blocks.

Maybe the following refactoring is a little bit better in terms of code readability
```
final Property boundProperty = propertyBinder.makePropertyAndBind();
if ( property.isAnnotationPresent( JoinTable.class ) ) {
    boundProperty.setOptional( optional );
}
```
Not a big deal and feel free to ignore my comment.

Reply:
Thanks @NathanQingyangXu , I have applied your suggestion.